pippy/PipelineSchedule.py

```diff
@@ -19,10 +19,8 @@
 
 class PipelineStage(ABC, nn.Module):
     @abstractmethod
-    def forward(self, microbatch):
+    def forward(self, args: List[torch.tensor]) -> torch.tensor:
```

Comment:
Is it possible to support `*args` and `**kwargs` for more general use cases?

Reply:
@fegin Yes, definitely possible. Right now the component that is actually calling `forward()` is the `PipelineSchedule`, for example for gpipe [here](https://github.com/pytorch/PiPPy/blob/main/pippy/PipelineSchedule.py#L247). As you can see the forward is taking in a single microbatch. So how the `forward()` of PipelineStage looks is going to depend on how/if we want to enforce the structure of the microbatches to be. This is still TBD, but I expect I will make further changes after this.